packages/dapp/src/components/ListingHistory.tsx

```diff
@@ -0,0 +1,131 @@
+import * as React from "react";
+import styled from "styled-components";
+import { Civil } from "@joincivil/core";
+import { List } from "immutable";
+import { Subscription } from "rxjs";
+
+import ListingEvent from "./ListingEvent";
+
+const StyledDiv = styled.div`
+  display: flex;
+  flex-wrap: wrap;
+  width: 100%
+  color: black;
+`;
+
+export interface ListingHistoryProps {
+  match: any;
+}
+
+export interface ListingHistoryState {
+  listingHistory: List<any>;
+  applicationSubscription: Subscription;
+  whitelistSubscription: Subscription;
+  readyToWhitelistSubscription: Subscription;
+  inChallengeCommitSubscription: Subscription;
+  inChallengeRevealSubscription: Subscription;
+  canBeUpdatedSubscription: Subscription;
+  awaitingAppealRequestSubscription: Subscription;
+  awaitingAppealJudgmentSubscription: Subscription;
+  error: undefined | string;
+}
+
+class ListingHistory extends React.Component<ListingHistoryProps, ListingHistoryState> {
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      listingHistory: List<any>(),
+      applicationSubscription: new Subscription(),
+      whitelistSubscription: new Subscription(),
+      readyToWhitelistSubscription: new Subscription(),
+      inChallengeCommitSubscription: new Subscription(),
+      inChallengeRevealSubscription: new Subscription(),
+      canBeUpdatedSubscription: new Subscription(),
+      awaitingAppealRequestSubscription: new Subscription(),
+      awaitingAppealJudgmentSubscription: new Subscription(),
+      error: undefined,
+    };
+  }
+
+  public componentWillMount(): void {
+    window.addEventListener("load", this.initListings);
+  }
+
+  public componentWillUnmount(): void {
+    this.state.applicationSubscription.unsubscribe();
+    this.state.whitelistSubscription.unsubscribe();
+    this.state.readyToWhitelistSubscription.unsubscribe();
+    this.state.inChallengeCommitSubscription.unsubscribe();
+    this.state.inChallengeRevealSubscription.unsubscribe();
+    this.state.canBeUpdatedSubscription.unsubscribe();
+    this.state.awaitingAppealRequestSubscription.unsubscribe();
+    this.state.awaitingAppealJudgmentSubscription.unsubscribe();
+    window.removeEventListener("load", this.initListings);
+  }
+
+  public render(): JSX.Element {
+    return (
+      <StyledDiv>
+        History:<br />
+        {this.state.listingHistory.map(e => {
+          return <ListingEvent key={this.props.match.params.listing + e.blockNumber} event={e} />;
+        })}
+      </StyledDiv>
+    );
+  }
+
+  // TODO(nickreynolds): move this all into redux
+  private initListings = async () => {
+    const civil = new Civil();
```

Comment:
Hum, this also might fail when there's not metamask. Ought to make the constructor safe